sopel/cli/config.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+"""Sopel Config Command Line Interfaces (CLI): ``sopel-config``"""
+from __future__ import unicode_literals, absolute_import, print_function, division
+
+import argparse
+import os
+
+from sopel import tools, config
+from . import utils
+
+
+def build_parser():
+    """Configure an argument parser for ``sopel-config``"""
+    parser = argparse.ArgumentParser(
+        description='Sopel configuration tool')
+
+    # Subparser: sopel-config <sub-parser> <sub-options>
+    subparsers = parser.add_subparsers(
+        help='Actions to perform',
+        dest='action')
+
+    # sopel-config list
+    list_parser = subparsers.add_parser(
+        'list',
+        help='List available configurations from Sopel\'s homedir',
+        description="""
+        List available configurations from Sopel's homedir ({homedir}) with the
+        extension "{ext}".
+        """.format(homedir=config.DEFAULT_HOMEDIR, ext='.cfg'))
+    list_parser.add_argument(
+        '-e', '--ext', '--extension',
+        dest='extension',
+        default='.cfg',
+        help='Filter by extension (default to "%(default)s)"')
+    list_parser.add_argument(
+        '-p', '--path',
+        action='store_true',
+        dest='display_path',
+        default=False,
+        help='Display a list of absolute filenames instead of their names')
+
+    # sopel-config init
+    init_parser = subparsers.add_parser(
+        'init',
+        help='Initialize sopel configuration file',
+        description='Initialize sopel configuration file')
+    utils.add_common_arguments(init_parser)
+
+    # sopel-config get <section> <key>
+    get_parser = subparsers.add_parser(
+        'get',
+        help='Get a configuration option\'s value',
+        description='Get a configuration option\'s value',
+    )
+    get_parser.add_argument('section')
+    get_parser.add_argument('option')
+    utils.add_common_arguments(get_parser)
+
+    return parser
+
+
+def handle_list(options):
+    """Display a list of configuration available from Sopel's homedir
+
+    :param options: argument parser's parsed option
+
+    This command display an unordered list of config's name from the default
+    Sopel's homedir, without their extension::
```

Comment:
```suggestion
    default homedir, without their extensions::
```

(meant to be applied together with the previous line's suggestion)